test/Microsoft.Framework.DependencyInjection.Tests/ServiceProviderExtensionsTest.cs

```diff
@@ -55,6 +55,20 @@ public void GetServicesReturns_A_Service()
         }
 
         [Fact]
+        public void NonGeneric_GetServices_Returns_A_Service()
```

Comment:
Could we have tests that verify that multiple `IFoo` values are returned and we that we filter correctly - so that asking for `typeof(IDifferentFoo)` does not return `IFoo` interfaces?